go/controller/internal/autogen/autogen_api_translator.go

```diff
@@ -1038,6 +1039,14 @@ func (a *apiTranslator) createModelClientForProvider(ctx context.Context, modelC
 			}
 		}
 
+		if len(modelConfig.Spec.DefaultHeaders) > 0 {
```

Comment:
this here is now only done for the Ollama case -- it should be in each of the case statements,  right?